### PR TITLE
Build CRUD api

### PR DESCRIPTION
**Pull request by Amershabou**
@dclay1983 @jiayitang0106 Let me know what you think!

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ class App extends React.Component {
     const restaurantId = window.location.href.slice(36);
     $.ajax({
       type: 'GET',
-      url: 'api/restaurants/'+restaurantId,
+      url: 'localhost:3005/api/restaurants/'+restaurantId,
```
Comment (dclay1983) [inline client/components/App.jsx]: by removing `localhost,` you have fewer problems with express. It uses the server that served the static files when you do not include it.

File: server/app.js [hunk 2]
```diff
@@ -1,22 +1,59 @@
 const express = require('express');
-const bodyParser = require('body-parser');
+const parser = require('body-parser');
```
Comment (dclay1983) [inline server/app.js:2]: You could remove body parser altogether. You are already using express.json which is the built-in parser that works the same. You could then remove a dependeny and reduce load times.

File: server/app.js [hunk 3]
```diff
@@ -1,22 +1,59 @@
...
+  db.deleteRec(id)
+    .then(() => res.status(200).send('The record has been deleted!'))
+    .catch(err => {
+      console.log(err);
+      res.status(400).end();
+    });
+});
+
+
+const port = process.env.PORT || 3005;
```
Comment (dclay1983) [inline server/app.js:58]: Nice use of global variables first.

File: server/db/recs.js [hunk 4]
```diff
@@ -9,6 +9,6 @@ const recsSchema = new mongoose.Schema({
   recs: []
 });
 
-const Recs = mongoose.model('Recommendations', recsSchema);
+const Rec = mongoose.model('Rec', recsSchema);
```
Comment (dclay1983) [inline server/db/recs.js:12]: Could we also make the model schema not be plural? Not sure if this is an option, but it was a good catch in the first place.

**Review by dclay1983 — COMMENTED** [review]
Nice work, don't forget to update the README.md to document the API routes.

File: server/db/recs.js (hunk 4 above)
Comment (Amershabou) [reply server/db/recs.js:12]: Yes normally because the database will make it plural anyway.

File: server/app.js (hunk 3 above)
Comment (Amershabou) [reply server/app.js:58]: Thanks!

File: server/app.js (hunk 2 above)
Comment (Amershabou) [reply server/app.js:2]: Agreed! I usually use express for parsing but i think we need parser for app.use(parser.urlencoded({ extended: true }));

Thanks

File: client/components/App.jsx (hunk 1 above)
Comment (Amershabou) [reply client/components/App.jsx]: Agree! I'll change that.

**Pull request merged**